presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergPageSourceProvider.java

```diff
@@ -254,7 +254,8 @@ private static ConnectorPageSource createOrcPageSource(
                     inputStream,
                     stats);
 
-            OrcReader reader = new OrcReader(orcDataSource, options);
+            OrcReader reader = OrcReader.createOrcReader(orcDataSource, options)
+                    .orElseThrow(() -> new PrestoException(ICEBERG_MISSING_DATA, "File is empty"));
```

Comment:
`MISSING_DATA` is used for HDFS `BlockMissingException`. This is like a corrupt file or something. How about
```java
.orElseThrow(() -> new PrestoException(ICEBERG_BAD_DATA, "ORC file is zero length"));